guardrails/document_store.py

```diff
@@ -0,0 +1,82 @@
+import uuid
+from abc import ABC, abstractmethod
+from dataclasses import dataclass
+from typing import Any, Dict, List, Tuple
+
+from guardrails.embedding import EmbeddingBase
+from guardrails.vectordb import VectorDBBase
+
+
+@dataclass
+class Document:
+    id: str
+    pages: Dict[int, str]
+    metadata: Dict[Any, Any] = None
+
+
+@dataclass
+class Page:
+    num: int
+    text: str
+    doc_id: int
+    metadata: Dict[Any, Any]
+
+
+class DocumentStoreBase(ABC):
+    @abstractmethod
+    def add_document(self, document: Document):
+        ...
+
+    @abstractmethod
+    def search(self, query: str, k: int = 4) -> List[Page]:
+        ...
+
+
+class PageLocator:
+    def __init__(self):
+        self._id_to_doc_page: Dict[int, (int, int)] = {}
```

Comment:
Instead of an `int` -> `(int, int)` mapping, wdyt about an `int` -> `Page` mapping? Since `Page` already contains the `doc_id`.